From 55f414909e6d4e2e9cdcf7e5e81adbc9e6deaef5 Mon Sep 17 00:00:00 2001
From: "Russ W. Knize" <Russ.Knize@motorola.com>
Date: Tue, 15 Sep 2015 16:50:42 -0500
Subject: [PATCH 353/959] IKSWM-3341: esdfs: parse the Android user ID from the
 correct root

When in "legacy" or "multi" derived permissions modes, look for the user
directory at the legacy root level, not at the normal root level.

Change-Id: I4144e667b6cffd14c52981159117fb2407ff5fe8
Signed-off-by: Russ Knize <Russ.Knize@motorola.com>
Reviewed-on: http://gerrit.mot.com/783749
SME-Granted: SME Approvals Granted
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Igor Kovalenko <igork@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 fs/esdfs/derive.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/esdfs/derive.c b/fs/esdfs/derive.c
index f9a83061bc9..5a3e0c75017 100644
--- a/fs/esdfs/derive.c
+++ b/fs/esdfs/derive.c
@@ -305,6 +305,7 @@ void esdfs_derive_perms(struct dentry *dentry)
 	switch (inode_i->tree) {
 	case ESDFS_TREE_ROOT_LEGACY:
 		inode_i->tree = ESDFS_TREE_ROOT;
+		ret = kstrtou32(dentry->d_name.name, 0, &inode_i->userid);
 		if (!strncasecmp(dentry->d_name.name,
 					"obb",
 					dentry->d_name.len))
@@ -313,7 +314,6 @@ void esdfs_derive_perms(struct dentry *dentry)
 
 	case ESDFS_TREE_ROOT:
 		inode_i->tree = ESDFS_TREE_MEDIA;
-		ret = kstrtou32(dentry->d_name.name, 0, &inode_i->userid);
 		if (!strncasecmp(dentry->d_name.name,
 				 "Android",
 				 dentry->d_name.len))
-- 
2.11.0

